ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/manager/task/AbstractEthTask.java

```diff
@@ -123,22 +126,16 @@ public final boolean isCancelled() {
   /**
    * Utility for registring completable futures for cleanup if this EthTask is cancelled.
    *
-   * @param subTaskFuture the future to be reigstered.
    * @param <S> the type of data returned from the CompletableFuture
-   * @return The completableFuture that was executed
+   * @param subTaskFuture the future to be registered.
    */
-  protected final <S> CompletableFuture<S> registerSubTask(
-      final CompletableFuture<S> subTaskFuture) {
+  protected final <S> void registerSubTask(final CompletableFuture<S> subTaskFuture) {
     synchronized (result) {
       if (!isCancelled()) {
         subTaskFutures.add(subTaskFuture);
-        subTaskFuture.whenComplete(
-            (r, t) -> {
-              subTaskFutures.remove(subTaskFuture);
-            });
-        return subTaskFuture;
+        subTaskFuture.whenComplete((r, t) -> subTaskFutures.remove(subTaskFuture));
       } else {
-        return completedExceptionally(new CancellationException());
+        completedExceptionally(new CancellationException());
```

Comment:
This creates a new future that is then immediately thrown away so I think this whole else block can just be removed.

Reply:
Done.